src/main/java/com/google/sps/data/ForumElement.java

```diff
@@ -0,0 +1,24 @@
+package com.google.sps.data;
+
+/** A comment or question element of the forum. */
+public final class ForumElement {
+
+  private final long id;
+  private final long parentId;
+  private final String topic;
+  private final long timestamp;
+  private final long likes;
+  private final String text;
+  // private final long userId
+
+  public ForumElement(
```

Comment:
As I commented in Praneet's PR, a factory is usually better than constructor with many parameters: https://github.com/googleinterns/granite2020/pull/12#discussion_r466440644

Here the factory would take Entity, which would make the servlet code cleaner, as it's the responsibility of this class to create itself.

Reply:
I created the factory method.  Should that replace the constructor?  The factory method uses my constructor now that is private.

Reply:
This is perfect now. You always need a constructor, but a private constructor only called from the factory is the exact pattern we're after.